packages/flutter_tools/lib/src/build_system/targets/macos.dart

```diff
@@ -343,3 +449,64 @@ class DebugBundleFlutterAssets extends Target {
     Source.pattern('{PROJECT_DIR}/macos/Flutter/ephemeral/App.framework/Versions/A/Resources/flutter_assets/isolate_snapshot_data'),
   ];
 }
+
+/// Bundle the profile flutter assets into the App.framework.
+class ProfileMacOSBundleFlutterAssets extends MacOSBundleFlutterAssets {
+  const ProfileMacOSBundleFlutterAssets();
+
+  @override
+  String get name => 'profile_macos_bundle_flutter_assets';
+
+  @override
+  List<Target> get dependencies => const <Target>[
+    CompileMacOSFramework(),
+    ProfileUnpackMacOS(),
+  ];
+
+  @override
+  List<Source> get inputs => const <Source>[
+    Source.pattern('{PROJECT_DIR}/pubspec.yaml'),
+    Source.behavior(MacOSAssetBehavior())
+  ];
+
+  @override
+  List<Source> get outputs => const <Source>[
+    Source.behavior(MacOSAssetBehavior()),
+    Source.pattern('{PROJECT_DIR}/macos/Flutter/ephemeral/App.framework/Versions/A/App'),
+    Source.pattern('{PROJECT_DIR}/macos/Flutter/ephemeral/App.framework/Versions/A/Resources/Info.plist'),
+    Source.pattern('{PROJECT_DIR}/macos/Flutter/ephemeral/App.framework/Versions/A/Resources/flutter_assets/AssetManifest.json'),
+    Source.pattern('{PROJECT_DIR}/macos/Flutter/ephemeral/App.framework/Versions/A/Resources/flutter_assets/FontManifest.json'),
+    Source.pattern('{PROJECT_DIR}/macos/Flutter/ephemeral/App.framework/Versions/A/Resources/flutter_assets/LICENSE'),
```

Comment:
Similar comment here; whatever approach is used for de-duping shared source elements in the other class tree should be used here for all the things that are the same, instead of listing all the duplicated elements three times.